cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -345,12 +344,16 @@ private Action parseAction(String action) {
 
   private void performUpgrade() throws Exception {
     LOG.info("Upgrading System and User Datasets ...");
+    HBaseAdmin hBaseAdmin = new HBaseAdmin(hConf);
     DatasetUpgrader dsUpgrade = injector.getInstance(DatasetUpgrader.class);
     dsUpgrade.upgrade();
+    hBaseTableUtil.dropTable(hBaseAdmin, dsUpgrade.getDatasetInstanceMDSUpgrader().getOldDatasetInstanceTableId());
```

Comment:
if dropping tables right after each upgrade method, why not drop them within the Upgrader class?